apis/kueue/v1alpha1/multikueueconfig_types.go

```diff
@@ -62,21 +50,70 @@ type KubeconfigRef struct {
 	LocationType LocationType `json:"locationType"`
 }
 
-//+genclient
-//+genclient:nonNamespaced
-//+kubebuilder:object:root=true
-//+kubebuilder:storageversion
-//+kubebuilder:resource:scope=Cluster
+type MultiKueueClusterSpec struct {
+	// Information how to connect to the cluster.
+	KubeConfig KubeConfig `json:"kubeConfig"`
+}
+
+type MultiKueueClusterStatus struct {
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	// +patchStrategy=merge
+	// +patchMergeKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+
+// MultiKueueCluster is the Schema for the multikueue API
+type MultiKueueCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   MultiKueueClusterSpec   `json:"spec,omitempty"`
+	Status MultiKueueClusterStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// MultiKueueClusterList contains a list of MultiKueueCluster
+type MultiKueueClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []MultiKueueCluster `json:"items"`
+}
+
+// MultiKueueConfigSpec defines the desired state of MultiKueueConfig
+type MultiKueueConfigSpec struct {
+	// List of MultiKueueClusters names where the workloads from the ClusterQueue should be distributed.
+	//
+	// +listType=set
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=100
```

Comment:
I would be surprised if MK can scale that much. Let's start more conservative (10) and we can always relax validation in the future.